src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,39 +1,66 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  modalTodo: Todo;
+  setModalTodo: (todo: Todo | null) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ modalTodo, setModalTodo }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    getUser(modalTodo.userId)
+      .then(setUser)
+      .finally(() => setIsLoading(false));
+  }, [modalTodo.userId]);
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
+      {isLoading ? (
         <Loader />
       ) : (
-        <div className="modal-card">
+        <div className="modal-card" data-cy="loader">
```

Comment:
`data-cy="loader"` should be only for the loader, probably that's why the test fails